internal/tofu/graph.go

```diff
@@ -44,11 +44,13 @@ func (g *Graph) walk(walker GraphWalker) tfdiags.Diagnostics {
 	// The callbacks for enter/exiting a graph
 	ctx := walker.EvalContext()
 
+	panicHandler := logging.PanicHandlerWithTraceFn()
```

Comment:
Note: This panicHandler (and call to debug.Stack()) only happens once, and not for each vertex within the graph.  This means the performance impact should be negligible.